lab_2_retrieval_w_bm25/main.py

```diff
@@ -168,6 +275,34 @@ def calculate_bm25_with_cutoff(
 
     In case of corrupt input arguments, None is returned.
     """
+    is_not_correct = (not vocab or not isinstance(vocab, list) or
+                      not all(isinstance(i, str) for i in vocab) or
+                      not document or not isinstance(document, list) or
+                      not all(isinstance(i, str) for i in document) or
+                      not idf_document or not isinstance(idf_document, dict) or
+                      not all((isinstance(key, str) and isinstance(value, float)
+                               for key, value in idf_document.items())) or
+                      all(isinstance(i, dict) for i in idf_document.values()) or
+                      not alpha or not isinstance(alpha, float) or
+                      not k1 or not isinstance(k1, float) or not 1.2 <= k1 <= 2.0 or
```

Comment:
magic numbers, Where did you find it?

Reply:
In the documentation
![image](https://github.com/user-attachments/assets/abd54462-712f-4bbe-8ee4-b8a8e4ddbc73)

Reply:
But I deleted it anyway